src/components/checkbox/checkbox.ts

```diff
@@ -248,6 +215,32 @@ export class MdCheckbox implements ControlValueAccessor {
     }
   }
 
+  /*
+   * We use a hidden native input field to handle changes to focus state via keyboard navigation,
+   * with visual rendering done separately. The native element is kept in sync with the overall
+   * state of the component.
+   */
+  onInputFocus() {
+    this.hasFocus = true;
+  }
+
+  onInputBlur() {
+    this.hasFocus = false;
+  }
+
+  /**
+   * Event handler for checkbox input element.  Toggles checked state if element is not disabled.
+   * @param event
+   */
+  onInteractionEvent(event: Event) {
+    if (this.disabled) {
+      event.stopPropagation();
+    } else {
+      this.checked = !this.checked;
+      this._elementRef.nativeElement.querySelector('input').focus();
```

Comment:
Actually, now that I implemented the label-based solution, this isn't necessary at all.